### PR TITLE
Refactor dependency-watchdog Helm chart into component

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity quality robustness
/kind technical-debt test
/priority 3
/topology seed
/platform all
/exp intermediate
/merge squash

**What this PR does / why we need it**:
This PR refactors the deployment for the `dependency-watchdog`. It adopts the component concept. Consequently, it eliminates the Helm chart and adds unit tests, instead. This improves several aspects and the overall code quality.

**Which issue(s) this PR fixes**:
Part of #2754

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/operation/botanist/component/kubeapiserver/dependency_watchdog.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+							{
+								Key:      v1beta1constants.LabelRole,
+								Operator: metav1.LabelSelectorOpNotIn,
+								Values:   []string{v1beta1constants.ETCDRoleMain},
+							},
+							{
+								Key:      v1beta1constants.LabelRole,
+								Operator: metav1.LabelSelectorOpNotIn,
+								Values:   []string{v1beta1constants.LabelAPIServer},
+							},
```
Comment (vpnachev) [inline pkg/operation/botanist/component/kubeapiserver/dependency_watchdog.go]: What do you think about:
```suggestion
								Values:   []string{v1beta1constants.ETCDRoleMain, v1beta1constants.LabelAPIServer},
							},
```

It was not configured like this previously, but should have the same effect.

**Review by vpnachev — DISMISSED** [review]
/lgtm

Just one question/proposal.

File: pkg/operation/botanist/component/kubeapiserver/dependency_watchdog.go (hunk 1 above)
Comment (rfranzke) [reply pkg/operation/botanist/component/kubeapiserver/dependency_watchdog.go]: Yes, it should. @amshuman-kr any comments? Why wasn't it done that way right away?

Comment (amshuman-kr) [reply pkg/operation/botanist/component/kubeapiserver/dependency_watchdog.go]: @vpnachev's suggestion is equivalent and more compact. I am not sure why it was done using two separate expression entries before.
I will check the rest of the PR later today. Sorry for the delay.

**Review by vpnachev — DISMISSED** [review]
/lgtm

File: pkg/operation/botanist/component/dependencywatchdog/dependency_watchdog.go [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+				},
+				UpdatePolicy: &autoscalingv1beta2.PodUpdatePolicy{
+					UpdateMode: &updateMode,
+				},
+				ResourcePolicy: &autoscalingv1beta2.PodResourcePolicy{
+					ContainerPolicies: []autoscalingv1beta2.ContainerResourcePolicy{{
+						ContainerName: autoscalingv1beta2.DefaultContainerResourcePolicy,
+						MinAllowed: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("25m"),
+							corev1.ResourceMemory: resource.MustParse("25Mi"),
```
Comment (amshuman-kr) [inline pkg/operation/botanist/component/dependencywatchdog/dependency_watchdog.go]: The probe uses a larger cache and needs a bit more memory, I think. (https://github.com/gardener/gardener/pull/4258/files#diff-aeb5d90a9661185c01f58b35703252e2f1347f1a396f98d690601c7ec9eab83dL75)

**Review by amshuman-kr — COMMENTED** [review]
Thanks @rfranzke! Just one small question about the memory `minAllowed` VPA configuration. Otherwise LGTM.

File: pkg/operation/botanist/component/dependencywatchdog/dependency_watchdog.go (hunk 2 above)
Comment (rfranzke) [reply pkg/operation/botanist/component/dependencywatchdog/dependency_watchdog.go]: Nice catch, fixed in 559d65b

**Review by vpnachev — DISMISSED** [review]
/lgtm

**Review by vpnachev — DISMISSED** [review]
/lgtm

**Review by amshuman-kr — DISMISSED** [review]
Thanks for the changes @rfranzke. /lgtm

**Comment by timebertt** [comment]
@rfranzke there are some `make check` errors.

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**